### DISCUSSION TITLE
Need  Updated One Questionset Response Format

### DISCUSSION THREAD

**Discussion by jayesh9747** [Sunbird-inQuiry/Community#114] [Documentation Issues]
I need an updated Question Set Response Format.
I am working on the migration of a questionnaire from the ML Survey service to the Inquiry Building Block. This part is complete, but for consumption, I need the Question Set Response format. Could you please share a sample question set response?

**Comment by jayesh9747** [comment]
@pallakartheekreddy , can you please share ? @krgauraw  sit told me you will help me for this.

**Reply by jayesh9747** [reply]
@rajnishdargan

**Reply by rajnishdargan** [reply]
Hi @jayesh9747 
Could you please clarify exactly what you're looking for? Are you asking for the latest QuML 1.1 schema for questions and question sets?
[Question Set Schema (QuML 1.1)](https://github.com/Sunbird-inQuiry/inquiry-api-service/blob/release-8.0.0/schemas/questionset/1.1/schema.json)
[Question Schema (QuML 1.1)](https://github.com/Sunbird-inQuiry/inquiry-api-service/blob/release-8.0.0/schemas/question/1.1/schema.json)

Or are you looking for a sample question set to see how the data is structured? Here’s a recent example created on Sunbird staging:
[Sample Question Set](https://staging.sunbirded.org/play/questionset/do_21400092794588364811) (No login required)
You can check the network tab for the question set hierarchy read API call.

Does this information meet your needs?

**Reply by jayesh9747** [reply]
is staging server is running ? whenever i hit the server it gives me the 502 error

**Reply by jayesh9747** [reply]
i share some screen shot and explain in details.

**Reply by jayesh9747** [reply]
![image](https://github.com/user-attachments/assets/7c919293-bae3-499e-8267-55bac7201c84)
I am working on the migration of questions and question sets. Initially, the staging server was running, but it went down afterward. To continue, I created a mock server and used the responses provided in the documentation. However, I really need the actual format of the question set hierarchy so I can transform the hierarchy into the current format of ML-Survey submission. This is why I need the API response. If you have a local setup or a hosted test server, please share it with me.

**Reply by jayesh9747** [reply]
@rajnishdargan please share me as soon as possible.

**Reply by rajnishdargan** [reply]
@jayesh9747 
I'm not sure if dock staging is working or if it's down. You might need to reach out to the support team to check on this.

@pallakartheekreddy , could you please confirm if dock staging is currently in use or if it's down?

In the meantime, @jayesh9747 , you can use the Sunbird staging for reading the questionset. While I don't have login access, you can still try using it for consumption without logging in.

Here’s one of the V1 questionsets: https://staging.sunbirded.org/play/questionset/do_214011731688161280120, although I previously shared a V2 questionset. See if this works for you.

I'm not familiar with ML-Survey. @pallakartheekreddy , could you assist Jayesh further if needed?

cc @krgauraw

**Reply by jayesh9747** [reply]
@rajnishdargan  can you send the response of Questionset hierarchy ?

**Reply by pallakartheekreddy** [reply]
@rajnishdargan Yes, Dock staging is down. Currently, we have only sunbird staging up and running.

**Reply by jayesh9747** [reply]
@pallakartheekreddy  can you send one response of Questionset hierarchy ?

**Reply by jayesh9747** [reply]
I have staging credentials but don't have permission for consumption. Can you grant me this privilege?
![image](https://github.com/user-attachments/assets/8fa91348-6728-4af8-84ac-152e8202f0f8)

**Reply by jayesh9747** [reply]
@vaivk369

**Reply by pallakartheekreddy** [reply]
@jayesh9747 share the Question set do id

**Reply by jayesh9747** [reply]
do_214103529528164352117  this question set is created in dockstaging server.  i want to question set hierarchy.

**Reply by pallakartheekreddy** [reply]
@jayesh9747 As I mentioned, we don't have a dock staging server now. I can't access the above Question set.

**Comment by jayesh9747** [comment]
@pallakartheekreddy then you share any questionset  hierarchy response  or grant the privilege to ' cbsestaging1@yopmail.com '  and 'client_id: project-sunbird-stage-client' to consume this api .

**Reply by pallakartheekreddy** [reply]
@jayesh9747 Please find below sample question set response for your reference 
[do_214149577796968448137.json](https://github.com/user-attachments/files/17125761/do_214149577796968448137.json)

**Reply by pallakartheekreddy** [reply]
@jayesh9747 Hope your query is now resolved. Please mark the thread as answered

**Reply by jayesh9747** [reply]
@pallakartheekreddy  can you grant the privilege to create the question and  questionset to [cbsestaging1@yopmail.com](mailto:cbsestaging1@yopmail.com)

**Comment by jayesh9747** [comment]
@pallakartheekreddy @rajnishdargan  Can you tell me which role has the privilege to create questions and question sets, as well as to read the question set hierarchy?
